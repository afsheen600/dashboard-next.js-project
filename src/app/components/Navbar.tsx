"use client";

import { AppBar, Toolbar, IconButton, Typography } from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import { useSidebar } from "../../context/SidebarContext";

const Navbar: React.FC = () => {
  const { toggleSidebar }: { toggleSidebar: () => void } = useSidebar();

  return (
    <AppBar
      position="fixed"
      sx={{ zIndex: (theme) => theme.zIndex.drawer + 1 }}
    >
      <Toolbar>
        <IconButton
          size="large"
          edge="start"
          color="inherit"
          aria-label="menu"
          sx={{ mr: 2 }}
          onClick={toggleSidebar}
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
          Next.js App
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

export default Navbar;
