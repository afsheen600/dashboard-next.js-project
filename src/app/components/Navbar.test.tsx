import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Navbar from "./Navbar";

// Mock useSidebar context
const toggleSidebarMock = jest.fn();
jest.mock("../../context/SidebarContext", () => ({
  useSidebar: () => ({
    toggleSidebar: toggleSidebarMock,
  }),
}));

describe("Navbar", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders the app title", () => {
    render(<Navbar />);
    expect(screen.getByText(/next\.js app/i)).toBeInTheDocument();
  });

  it("renders the menu icon button with correct aria-label", () => {
    render(<Navbar />);
    const menuButton = screen.getByLabelText(/menu/i);
    expect(menuButton).toBeInTheDocument();
    expect(menuButton).toHaveAttribute("aria-label", "menu");
  });

  it("calls toggleSidebar when menu button is clicked", () => {
    render(<Navbar />);
    const menuButton = screen.getByLabelText(/menu/i);
    fireEvent.click(menuButton);
    expect(toggleSidebarMock).toHaveBeenCalledTimes(1);
  });

  it("menu button is of type button and has correct size", () => {
    render(<Navbar />);
    const menuButton = screen.getByLabelText(/menu/i);
    expect(menuButton).toHaveAttribute("type", "button");
    expect(menuButton).toHaveClass("MuiIconButton-sizeLarge");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
