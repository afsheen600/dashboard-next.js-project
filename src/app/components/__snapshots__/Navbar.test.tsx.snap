// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Navbar matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1wtssb0-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-w6cvuv-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
      </button>
      <div
        class="MuiTypography-root MuiTypography-h6 css-1bv5yfe-MuiTypography-root"
      >
        Next.js App
      </div>
    </div>
  </header>
</DocumentFragment>
`;
